src/modules/WithParamsAndPublicKeys.js

```diff
@@ -258,20 +252,20 @@ export default class WithParamsAndPublicKeys {
   }
 
   async getParams(did, counter) {
-    const hexId = getHexIdentifierFromDID(did);
+    const hexId = typedHexDID(this.api, did);
     return this.getParamsByHexDid(hexId, counter);
   }
 
   /**
    *
    * @param did
-   * @param keyId
+   * @param signingKeyRef
    * @param withParams - If true, return the params referenced by the public key. It will throw an error if paramsRef is null
    * or params were not found on chain which can happen if they were deleted after this public key was added.
    * @returns {Promise<{bytes: string}|null>}
    */
   async getPublicKey(did, keyId, withParams = false) {
-    const hexId = getHexIdentifierFromDID(did);
+    const hexId = typedHexDID(this.api, did);
```

Comment:
`did:key` can't have an onchain public key so you can just use `getHexIdentifier` and avoid creating extra object in `typedHexDID`